src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -336,6 +336,31 @@ internal Authority GetAuthority(IAccount account)
             return await handler.RunAsync(CancellationToken.None).ConfigureAwait(false);
         }
 
+        internal async Task<AuthenticationResult> ExchangeRefreshTokenAsync(string userProvidedRefreshToken)
+        {
+            var reqParams = new AuthenticationRequestParameters
+            {
+                SliceParameters = SliceParameters,
+                Authority = Instance.Authority.CreateAuthority(ServiceBundle, Authority, false),
+                ClientId = ClientId,
+                TokenCache = UserTokenCache,
+                Scope = new SortedSet<string>(),
+                RedirectUri = new Uri(RedirectUri),
+                RequestContext = CreateRequestContext(Guid.Empty),
+                ValidateAuthority = ValidateAuthority,
+                IsExtendedLifeTimeEnabled = ExtendedLifeTimeEnabled,
+                ExchangingRefreshToken = true
+            };
+
+            var handler = new ByRefreshTokenRequest(
+                ServiceBundle,
+                reqParams,
+                ApiEvent.ApiIds.AcquireTokenSilentWithoutAuthority,
```

Comment:
We should have a new ApiID since this is a new flow, so that we can collect telemetry about it. I believe it's ok to just add another member in that enum, but @sangonzal can confirm is anything else is needed on the "server (kusto) side"

Reply:
This does not seem resolved.